File: 0492-construct-the-rectangle/0492-construct-the-rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        if area==0:
            return [0,0]
        opWidth=1
        mindiff=math.inf
        for l in range(area,1,-1):
            if area%l!=0:
                continue
            w=area/l
            print(l,w)
            if l<w:
                break
            if l-w<mindiff:
                mindiff=l-w
                opWidth=w
        return [int(area/opWidth),int(opWidth)]

File: 0492-construct-the-rectangle/0492-construct-the-rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        int ans[]=new int[2];
        int j=(int)Math.sqrt(area);
        int l=1,w=1,diff=Integer.MAX_VALUE;
        for(int i=1;i<=j;i++){
            if (area%i==0){
                int a=i;
                int b=area/i;
                if (Math.abs(a-b)<diff){
                    diff=Math.abs(a-b);
                    l=Math.max(a,b);
                    w=Math.min(a,b);
                }
            }
        }
        ans[0]=l;
        ans[1]=w;
        return ans ;
    }
}